src/arch/arm/kernel/boot.c

```diff
@@ -626,5 +626,6 @@ BOOT_CODE VISIBLE void init_kernel(
 #endif
     schedule();
     activateThread();
+    restore_user_context();
+    UNREACHABLE();
```

Comment:
This is probably not a good idea for verification. `restore_user_context()` doesn't have a control flow that is expressible in C, so if we are ever going to verify kernel initialisation, we'll have to remove it again, or at least make it separate from the main `init_kernel` function.

Generally `restore_user_context()` is assumed and dealt with in the machine interface. The only time we do deal with it in C is in the fast path exit, but the fast path is already a very special case, and every new special case for something like this (proof assumptions) is a risk of getting something wrong.

This would have to be a blocker for right now, because kernel init verification is not in the "very soon" project bucket at the moment, but it would have to be changed again when we get there.